metadata-io/src/main/java/com/linkedin/metadata/system_info/collectors/SpringComponentsCollector.java

```diff
@@ -0,0 +1,266 @@
+package com.linkedin.metadata.system_info.collectors;
+
+import static com.linkedin.metadata.system_info.SystemInfoConstants.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.metadata.system_info.ComponentInfo;
+import com.linkedin.metadata.system_info.ComponentStatus;
+import com.linkedin.metadata.system_info.SpringComponentsInfo;
+import com.linkedin.metadata.version.GitVersion;
+import io.datahubproject.metadata.context.OperationContext;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Collects information about Spring components (GMS, MAE Consumer, MCE Consumer).
+ *
+ * <p>DEPLOYMENT MODES: DataHub can be deployed in two ways: 1. Embedded mode (default quickstart):
+ * All components run in single GMS container 2. Separate containers: Each component runs in its own
+ * container for better scaling
+ *
+ * <p>CURRENT LIMITATION: This collector currently returns placeholder data for MAE/MCE consumers to
+ * avoid a circular dependency issue that occurs in embedded mode:
```

Comment:
I'm not sure how the original PR intended to work, if I'm not mistaken it was trying to query the same endpoint and thus it was a circular dependency which likely resulted in timeouts. 

In any case, I kept placeholders for getting MAE and MCE and I kept fetchRemoteComponentInfo though it is not currently used, my indent was to preserve the original structure as much as possible.

Reply:
I can do MAE and MCE consumers as a separate pr. I would have separate, specialized endpoints to make the distinction more explicit. I would expect an instance would fail the method if it is not an MCE/MAE